src/generic/datavgen.cpp

```diff
@@ -2046,69 +2051,215 @@ bool wxDataViewMainWindow::EnableDropTarget( const wxDataFormat &format )
+            {
+                wxDataViewColumn* col = GetOwner()->GetColumnAt(i);
+                if (col->IsHidden())
+                    continue;   // skip it!
+
+                if (col == expander)
+                    break;
+
+                start_x += col->GetWidth();
+            }
+
+            const int expanderWidth = wxRendererNative::Get().GetExpanderSize(this).GetWidth();
+
+            int level = node->GetIndentLevel();
+
+            wxDataViewTreeNode* prevAscendNode = node;
+            wxDataViewTreeNode* ascendNode = node;
+            while (ascendNode != NULL)
+            {
+                *indentLevel = level + 1;   // [out]
+
+                if (m_owner->GetModel()->IsContainer(ascendNode->GetItem()))
+                    // Can be inserted
+                {
+                    int itemPosition = ascendNode->FindChildByItem(prevAscendNode->GetItem());
+                    *proposedDropIndex  = itemPosition == wxNOT_FOUND ? 0 : itemPosition + 1;   // [out]
+                    *item               = ascendNode->GetItem();                                // [out]
+
+                    if (IsExpanded(row) && itemPosition != ascendNode->GetChildNodes().size() - 1)
+                        break;
+
+                    int indent = GetOwner()->GetIndent()*level + expanderWidth;
+
+                    if (xx >= start_x + indent)
+                        break;
+                }
+
+                prevAscendNode = ascendNode;
+                ascendNode = ascendNode->GetParent();
+                --level;
+            }
+        }
+
+        if (insertBefore)
+            return 2;       // BEFORE
+        else if (insertAfter)
+            return 3;       // AFTER
+    }
 
-    if ( row < GetRowCount() && xx <= GetEndOfLastCol() )
-        item = GetItemByRow( row );
+    return 1;   // INSIDE
+}
+
+
+wxDragResult wxDataViewMainWindow::OnDragOver( wxDataFormat format, wxCoord x,
+                                               wxCoord y, wxDragResult def )
+{
+    wxDataViewItem  item;
+    unsigned int    row = 0;
+    int             proposedDropIndex = -1;
+
+    int nextDropHint = GetDropItemInfo(x, y, &row, &item, &proposedDropIndex, &m_dropHintIndentLevel);
 
     wxDataViewEvent event(wxEVT_DATAVIEW_ITEM_DROP_POSSIBLE, m_owner, item);
+    event.SetProposedDropIndex(proposedDropIndex);
     event.SetDataFormat( format );
     event.SetDropEffect( def );
-    if ( !m_owner->HandleWindowEvent( event ) || !event.IsAllowed() )
+
+    wxDragResult result = def;
+
+    if (m_owner->HandleWindowEvent(event) && event.IsAllowed())
+        // Processing handled event
+    {
+        result = event.GetDropEffect();
+        switch (result)
+        {
+            case wxDragNone:
+            case wxDragCancel:
+            case wxDragError:
+            {
+                RemoveDropHint();
+                return result;
+            }
+
+            default:
```

Comment:
It would be better to explicitly list the remaining enum elements (`wxDragCopy`, `wxDragMove`, `wxDragLink`) for the switch to be exhaustive.

Reply:
Should `default` case be removed?

Reply:
Yes, if all the existing enum elements are covered, there should be no `default` -- this will allow the compiler to warn us if we need to add new cases if we ever add more elements to this enum.